packages/studioCMS/src/integrations/studioCMSDashboard/routes/databaseSetup/setup.ts

```diff
@@ -0,0 +1,175 @@
+// @ts-expect-error - Some types can only be imported from the Astro runtime
+import { Blog, Permissions, Pages, SiteConfig, User, db, eq } from 'astro:db';
+import { scryptAsync } from "@noble/hashes/scrypt";
+
+import type { APIContext } from "astro";
+
+export async function POST(context: APIContext): Promise<Response> {
+	const formData = await context.request.formData();
+
+	const setupLocalAdmin = formData.get("local-setup")
+
+	if (setupLocalAdmin === "1") {
+		const username = formData.get("local-admin-name")?.toString();
+		// username must be between 3 ~ 31 characters, and only consists of lowercase letters, 0-9, -, and _
+		// keep in mind some database (e.g. mysql) are case insensitive
+		if (
+			typeof username !== "string" ||
+			username.length < 3 ||
+			username.length > 31 ||
+			!/^[a-z0-9_-]+$/.test(username)
+		) {
+			return new Response(
+				JSON.stringify({
+					error: "Invalid username"
+				}),
+				{
+					status: 400
+				}
+			);
+		}
+		const password = formData.get("local-admin-password");
+		if (typeof password !== "string" || password.length < 6 || password.length > 255) {
+			return new Response(
+				JSON.stringify({
+					error: "Invalid password"
+				}),
+				{
+					status: 400
+				}
+			);
+		}
+		const name = formData.get("local-admin-display-name");
+
+
+		const existingUser = await db.select().from(User).where(eq(User.username, username)).get()
+
+		if (existingUser) {
+			return new Response(
+				JSON.stringify({
+					error: "User Error"
+				}),
+				{
+					status: 400
+				}
+			);
+		
+		}
+		await db
+			.insert(User)
+			.values({
+				name: name as string,
+				username,
+			})
+
+		const newUser = await db.select().from(User).where(eq(User.username, username)).get();
+		const hashedPassword = await scryptAsync(password, newUser.id, { N: 2 ** 12, r: 8, p: 1, dkLen: 32 })
```

Comment:
I'm guessing you found this somewhere?? Probably worth a link to wherever it came from so we can find it.

Lots of magic numbers and unhelpful names of keys! So I have no idea what's happening

Reply:
i used information from [This](https://words.filippo.io/the-scrypt-parameters/) to understand what they are used for, plus the `@nobile/hases` & `lucia-auth` docs to understand what is considered Sane defaults.  I plan to add more customizablity to the way it works with the option for user to override, but with this default value it should be able to run on WorkerD without causing to big of an issue for smaller projects

Reply:
it would be nice to add a small comment about it tho, just for future reference.

Reply:
from @Adammatthiesen 

https://discord.com/channels/1222990091811819731/1222990091811819734/1239703703376892036